providers/oidc.go

```diff
@@ -101,19 +101,27 @@ func (p *OIDCProvider) EnrichSession(_ context.Context, s *sessions.SessionState
 	return nil
 }
 
-// ValidateSession checks that the session's IDToken is still valid
+// ValidateSession checks that the session's id_token or access_token (when a ValidateURL is configured) is still valid
 func (p *OIDCProvider) ValidateSession(ctx context.Context, s *sessions.SessionState) bool {
-	_, err := p.Verifier.Verify(ctx, s.IDToken)
-	if err != nil {
+	if s.IDToken == "" {
+		logger.Println("no new id token has been redeemed after refresh")
+		if !validateToken(ctx, p, s.AccessToken, makeOIDCHeader(s.AccessToken)) {
+			logger.Errorf("access token is invalid")
+			return false
+		}
+		return true
```

Comment:
Do we need the log here? Or should we just return?
```suggestion
		return validateToken(ctx, p, s.AccessToken, makeOIDCHeader(s.AccessToken))
```